src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/OSPlatformAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Base type for all platform-specific API attributes.
+    /// </summary>
+#pragma warning disable CS3015 // Type has no accessible constructors which use only CLS-compliant types
+    public abstract class OSPlatformAttribute : Attribute
```

Comment:
@terrajobst the `private` contractor causing CS3015 warning, suppressed for now, please let me know if you prefer any other options: 
1. use [CLSCompliant(false)] attribute for all attributes
2. keep the suppression
3. use internal protected instead

Reply:
Suppressing the warning on the constructor is fine.